apps/designer/app/designer/features/style-panel/shared/style-info.ts

```diff
@@ -23,31 +57,164 @@ export const getStyleSource = (
   return "preset";
 };
 
+const styleProperties = Object.keys(properties) as StyleProperty[];
+const inheritableProperties = new Set<string>();
+for (const [property, value] of Object.entries(properties)) {
+  if (value.inherited) {
+    inheritableProperties.add(property);
+  }
+}
+
+/**
+ * find all breakpoints which may affect current view
+ */
+export const getCascadedBreakpointIds = (
+  breakpoints: Breakpoint[],
+  selectedBreakpointId?: string
+) => {
+  const sortedBreakpoints = breakpoints
+    .slice()
+    .sort((a, b) => a.minWidth - b.minWidth);
+  const cascadedBreakpointIds: string[] = [];
+  for (const breakpoint of sortedBreakpoints) {
+    if (breakpoint.id === selectedBreakpointId) {
+      break;
+    }
+    cascadedBreakpointIds.push(breakpoint.id);
+  }
+  return cascadedBreakpointIds;
+};
+
+/**
+ * extract all styles from breakpoints
+ * affecting current view
+ */
+export const getCascadedInfo = (
+  cssRules: CssRule[],
+  cascadedBreakpointIds: string[]
+) => {
+  const styles = new Map<string, Style>();
+  for (const rule of cssRules) {
```

Comment:
Just optimization, `cssRules` probably already contains all rules sorted by breakpoints (not sure)
in that case, we can just check that `cascadedBreakpointIds.includes(rule.breakpoint)` and fill `cascadedStyle` directly here.

Reply:
Potentially it will be insertion order. So I would keep it and refactor once we migrate to normalized styles.